component/pyroscope/ebpf/ebpfspy/sd/target.go

```diff
@@ -0,0 +1,194 @@
+package sd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component/pyroscope/ebpf/ebpfspy/metrics"
+	lru "github.com/hashicorp/golang-lru/v2"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/agent/component/discovery"
+)
+
+const (
+	labelContainerID    = "__container_id__"
+	labelServiceName    = "service_name"
+	labelServiceNameK8s = "__meta_kubernetes_pod_annotation_pyroscope_io_service_name"
+	metricValue         = "process_cpu"
+)
+
+type Target struct {
+	labels labels.Labels
+
+	fingerprint           uint64
+	fingerprintCalculated bool
+}
+
+func NewTarget(cid containerID, target discovery.Target) (*Target, error) {
+	serviceName := target[labelServiceName]
+	if serviceName == "" {
+		serviceName = inferServiceName(target)
+	}
+
+	lset := make(map[string]string, len(target))
+	for k, v := range target {
+		if strings.HasPrefix(k, model.ReservedLabelPrefix) && k != labels.MetricName {
+			continue
+		}
+		lset[k] = v
+	}
+	if lset[labels.MetricName] == "" {
+		lset[labels.MetricName] = metricValue
+	}
+	if lset[labelServiceName] == "" {
+		lset[labelServiceName] = serviceName
+	}
+	if cid != "" {
+		lset[labelContainerID] = string(cid)
+	}
+	return &Target{
+		labels: labels.FromMap(lset),
+	}, nil
+}
+
+func inferServiceName(target discovery.Target) string {
+	k8sServiceName := target[labelServiceNameK8s]
+	if k8sServiceName != "" {
+		return k8sServiceName
+	}
+	k8sNamespace := target["__meta_kubernetes_namespace"]
+	k8sContainer := target["__meta_kubernetes_pod_container_name"]
+	if k8sNamespace != "" && k8sContainer != "" {
+		return fmt.Sprintf("ebpf/%s/%s", k8sNamespace, k8sContainer)
```

Comment:
I like the idea to add agent=ebpf label
I don't like the idea to remove the prefix because if we remove, the profiles will be merged for both ebfp and scraped profiles because of the same service_name and it may be confusing as to why I see 4 hours of data when I expect just 2 hours of data, the stacktraces will mostlikely differ nad not merge perfectly, so there will be 2 almost identical profiles. 
Coming from og pyroscope as a user I realy just want to do a one mouse click and get the correct and expected app. And now I have to always  keep in mind to query for 2 labels for both ebpf and scraped profiles

Reply:
>the profiles will be merged for both ebfp and scraped profiles because of the same service_name

Yes but this our problem, we should never have configured the same service name to begging with. We should use relabeling on our ebpf config to add the prefix, not make it default. WDYT ?